packages/router/src/Set.tsx

```diff
@@ -81,9 +81,8 @@ export function Set<WrapperProps>(props: SetProps<WrapperProps>) {
         const currentLocation =
           global.location.pathname + encodeURIComponent(global.location.search)
 
-        const unauthenticatedPath = routerState.routes.filter(
-          ({ name }) => unauthenticated === name
-        )[0]?.path
+        // We already have a check for !unauthenticated further up
+        const unauthenticatedPath = namedRoutes[unauthenticated || '']()
```

Comment:
Maybe we should throw if we cannot find a route that's reference by `unauthenticated`?

Reply:
Something along the lines of:

"An "unauthenticated" prop is set to  "${unauthenticated}," but we could not find a route with that name."